modules/assim.batch/R/autoburnin.R

```diff
@@ -1,123 +1,58 @@
-#' @param use.confidence Logical. If TRUE (default), use 95% confidence interval for Gelman Diagnostic.
+#' @param method Character string indicating method (default="moving.window". 
+#' No other options are currently implemented.)
+#' @param use.confidence Logical. If TRUE (default), use 95% confidence 
+#' interval for Gelman Diagnostic.
 #' If FALSE, use the point estimate.
-#' @param width Width of window for `rmw` method. Default = niter / 2
-#' @param njump Number of steps for `rmw` method. Default = 50 
-#' @param plotfile Filename for a PNG of the Gelman plot. Default = '/dev/null'
-#' @param ... Other parameters to `gelman.plot`
-#' @inheritParams coda::gelman.diag
+#' @param width Width of window for "moving.window" method. Default = niter / 2
+#' @param njump Number of steps for "moving.window" method. Default = 50
+#' @param ... Other parameters to methods
 #' 
 #' @details 
-#' rmw (Reverse moving window) method divides the chains into `njump` sections, each of size `width`, and loops over them starting from the last one until the first block that has not converged.
-#' The iteration starting the next block after that is the burnin.
-#' 
-#' `gelman.plot` method uses the `gelman.plot` function from the `coda` package to calculate the Gelman diagnostic cumulatively from the beginning of the chain.
-#' The iteration after the last time the `gelman.diag` exceeds the threshold for any of the parameters is the burnin.
+#' See "gelman_diag_mw".
 #' 
 #' @examples
-#'      library(coda)
-#'      data(line)
-#'      burnin <- getBurnin(line, threshold = 1.05, 
-#'                          method = 'gelman.plot')
+#'      z1 <- coda::mcmc(c(rnorm(2500, 5), rnorm(2500, 0)))
+#'      z2 <- coda::mcmc(c(rnorm(2500, -5), rnorm(2500, 0)))
+#'      z <- coda::mcmc.list(z1, z2)
+#'      burnin <- getBurnin(z, threshold = 1.05)
 #' @author Alexey Shiklomanov, Michael Dietze
 #' @export
-getBurnin <- function(jags_out, 
-                      method = "rmw", 
-                      threshold = 1.1,
+
+getBurnin <- function(jags_out,
+                      threshold = 1.1, 
                       use.confidence = TRUE,
-                      width = ceiling(coda::niter(jags_out)/2), 
-                      njump = 50,
-                      autoburnin = FALSE,
+                      method = "moving.window",
                       plotfile = "/dev/null",
                       ...) {
-  if (!coda::is.mcmc.list(jags_out)) jags_out <- makeMCMCList(jags_out)
-  if(coda::niter(jags_out) < 50) {
-    stop("Fewer than 50 iterations. Burnin will not be calculated")
-  }
-  if (method == "rmw"){
-    burnin <- getBurnin.rmw(jags_out, 
-                            width = width,
-                            njump = njump,
-                            threshold = threshold,
-                            use.confidence = use.confidence,
-                            ...)
-  } else if (method == "gelman.plot") {
```

Comment:
:+1: Added this back in https://github.com/PecanProject/pecan/pull/1175/commits/943a486540eb4237f6861dc1a540d7e408205431